src/operations/blas1_trees.hpp

```diff
@@ -538,6 +538,87 @@ AssignReduction<operator_t, lhs_t, rhs_t>::adjust_access_displacement() {
   rhs_.adjust_access_displacement();
 }
 
+template <typename operand_t>
+Rotg<operand_t>::Rotg(operand_t& _a, operand_t& _b, operand_t& _c, operand_t& _s) : a_{_a}, b_{_b}, c_{_c}, s_{_s} {}
+
+template <typename operand_t>
+SYCL_BLAS_INLINE typename Rotg<operand_t>::index_t
+Rotg<operand_t>::get_size() const {
+  return static_cast<Rotg<operand_t>::index_t>(1);
+}
+
+template <typename operand_t>
+SYCL_BLAS_INLINE typename Rotg<operand_t>::value_t Rotg<operand_t>::eval(
+    typename Rotg<operand_t>::index_t i) {
+
+  using zero = constant<value_t, const_val::zero>;
+  using one = constant<value_t, const_val::one>;
+
+  value_t &a_ref = a_.eval(i);
+  value_t &b_ref = b_.eval(i);
+  value_t &c_ref = c_.eval(i);
+  value_t &s_ref = s_.eval(i);
+
+  const value_t abs_a = AbsoluteValue::eval(a_ref);
+  const value_t abs_b = AbsoluteValue::eval(b_ref);
+  const value_t sigma =
+      abs_a > abs_b ? SignOperator::eval(a_ref): SignOperator::eval(b_ref);
+  const value_t r =
+      ProductOperator::eval(sigma, HypotenuseOperator::eval(a_ref, b_ref));
+
+
+  if (r == zero::value()) {
+    c_ref = one::value();
+    s_ref = zero::value();
+  } else {
+    c_ref = DivisionOperator::eval(a_ref, r);
```

Comment:
Do we need this if abs_1 >= abs_b? Can we defer calculating it?

Reply:
Not sure if I understand the suggestion. If abs_a >= abs_b, the only thing that changes is the z output (parameter b) and the sigma.  But we still need to calculate a, c and s output parameters.

Reply:
I missed that `c` and `s` where also returned.